apps/etterlatte-beregning/src/main/kotlin/beregning/grunnlag/BeregningsGrunnlagService.kt

```diff
@@ -22,13 +22,15 @@ class BeregningsGrunnlagService(
     ): Boolean = when {
         behandlingKlient.beregn(behandlingId, bruker, false) -> {
             val behandling = behandlingKlient.hentBehandling(behandlingId, bruker)
-            if (behandling.behandlingType == BehandlingType.REVURDERING) {
+            val kanLagreDetteGrunnlaget = if (behandling.behandlingType == BehandlingType.REVURDERING) {
                 // Her vil vi sjekke opp om det vi lagrer ned ikke er modifisert før virk på revurderingen
                 val forrigeIverksatte = behandlingKlient.hentSisteIverksatteBehandling(behandling.sak, bruker)
-                sjekkAtGrunnlagIkkeErEndretFoerVirk(behandling, forrigeIverksatte, barnepensjonBeregningsGrunnlag)
+                grunnlagErIkkeEndretFoerVirk(behandling, forrigeIverksatte, barnepensjonBeregningsGrunnlag)
+            } else {
+                true
             }
 
-            beregningsGrunnlagRepository.lagre(
+            kanLagreDetteGrunnlaget && beregningsGrunnlagRepository.lagre(
```

Comment:
Enhetstest for å sikre at dette blir korrekt?

Reply:
Ja, det er kanskje greit. Det er en del tester på `erGrunnlagLiktFoerDato`, som brukes i bunn, men jeg tenker det kan sikkert testes litt at det henger sammen her og.